### DISCUSSION TITLE
HashiCorp Credentials provider error when using -runner uber jar

### DISCUSSION THREAD

**Discussion by elmarm88** [quarkiverse/quarkus-vault#96] [General]
I have a small POC java app that has a db password stored in a HashiCorp vault.

I use it as a credentials provider in quarkus.

```
quarkus.datasource.soam.db-kind=oracle
quarkus.datasource.soam.jdbc.driver=oracle.jdbc.driver.OracleDriver
quarkus.datasource.soam.jdbc.url=jdbc:oracle:thin:@localhost:1521:CUSTOMD
quarkus.datasource.soam.username=soa_metadata
quarkus.vault.credentials-provider.soam.kv-path=myapps/vault-quickstart/soametadata
quarkus.datasource.soam.credentials-provider = soam
```
 
If I run the it from command line with

`mvn quarkus:dev
` 
it all works fine.

However if I run the -runner jar file in the /target directory (its an uber-jar) with the following command :

`java -jar poc-quarkus-vault-db-runner.jar`
 
I get the following error:

```
2022-12-06 12:23:59,870 WARN  [io.agroal.pool] (agroal-11) Datasource 'soam': Failed to create connection due to VaultClientException
2022-12-06 12:23:59,883 ERROR [sun.java.ws.rest.quarkus.poc.api.personinf.DbInfoService] (executor-thread-0) Error point 1: null: io.quarkus.vault.runtime.client.VaultClientException code=404 body={"errors":[]}
```
 
I should note that the app starts up fine, its just when I call the REST endpoint to pull something from the DB that is crashes.
Any ideas would be very welcome.

Thank you in advance

**Comment by kdubb** [comment]
@elmarm88 Was this solved? Seems it was missed by all of us.

**Comment by elmarm88** [comment]
Hi @kdubb  
I deleted the whole app and started from scratch and that seemed to do the trick. Think something got messed up somewhere but after spending 2 days bug hunting I decided on the Rip and Replace method :)
